internal/nitrite_test.go

```diff
@@ -0,0 +1,187 @@
+		},
+	}
+
+	for key := range attestatations {
+		t.Run("happy path", func(t *testing.T) {
+			t.Log(key)
+
+			// when
+			result, err := internal.Verify(
+				attestatations[key].attestation,
+				attestatations[key].certProvider,
+				internal.WithAttestationTime(),
+				attestatations[key].allowDebug,
+			)
+
+			// then
+			require.NoError(t, err)
+			// make sure at least one of the fields is populated and attested
+			assert.Equal(
+				t,
+				attestatations[key].time.UTC(),
+				result.Document.CreatedAt().UTC(),
+			)
+		})
+
+		t.Run("cannot get root certificates", func(t *testing.T) {
+			t.Log(key)
+
+			// given
+			certProvider := mocks.NewNitriteCertProvider(t)
+
+			// expecting
+			certProvider.EXPECT().Roots().Return(nil, assert.AnError)
+
+			// when
+			_, err := internal.Verify(
+				attestatations[key].attestation,
+				certProvider,
+				internal.WithAttestationTime(),
+				attestatations[key].allowDebug,
+			)
+
+			// then
+			assert.ErrorIs(t, err, assert.AnError)
+			assert.ErrorContains(t, err, "getting root certificates")
+		})
+
+		t.Run("unknown signing authority - nil roots", func(t *testing.T) {
+			t.Log(key)
+
+			// given
+			certProvider := mocks.NewNitriteCertProvider(t)
+
+			// expecting
+			certProvider.EXPECT().Roots().Return(nil, nil)
+
+			// when
+			_, err := internal.Verify(
+				attestatations[key].attestation,
+				certProvider,
+				internal.WithAttestationTime(),
+				attestatations[key].allowDebug,
+			)
+
+			// then
+			assert.ErrorContains(t, err, "verifying certificate")
+		})
+
+		t.Run("unknown signing authority - empty roots", func(t *testing.T) {
+			t.Log(key)
+
+			// given
+			certProvider := mocks.NewNitriteCertProvider(t)
+
+			// expecting
+			certProvider.EXPECT().Roots().Return(x509.NewCertPool(), nil)
+
+			// when
+			_, err := internal.Verify(
+				attestatations[key].attestation,
+				certProvider,
+				internal.WithAttestationTime(),
+				attestatations[key].allowDebug,
+			)
+
+			// then
+			assert.ErrorContains(t, err, "verifying certificate")
+		})
+
+		timeOutOfBoundsTests := []struct {
+			name        string
+			timeOpt     internal.VerificationTimeFunc
+			errContains string
+		}{
+			{
+				"certificate expired",
+				internal.WithTime(time.Date(10000, 0, 0, 0, 0, 0, 0, time.UTC)),
+				"verifying certificate",
+			},
+			{
+				"certificate not yet valid",
+				internal.WithTime(time.Date(1970, 0, 0, 0, 0, 0, 0, time.UTC)),
+				"verifying certificate",
+			},
+			{
+				"zero time",
+				internal.WithTime(time.Time{}),
+				"verification time is 0",
+			},
+		}
+		for _, tt := range timeOutOfBoundsTests {
+			t.Run(tt.name, func(t *testing.T) {
+				t.Log(key)
+
+				// when
+				_, err = internal.Verify(
+					attestatations[key].attestation,
+					attestatations[key].certProvider,
+					tt.timeOpt,
+					attestatations[key].allowDebug,
+				)
+
+				// then
+				assert.ErrorContains(t, err, tt.errContains)
+			})
+		}
+	}
```

Comment:
Could you add a single test case down here to check that we error out when trying to verify a debug attestation without debug turned on
```golang
	t.Run("attestation was generated in debug mode", func(t *testing.T) {
		// when
		_, err := internal.Verify(
			attestatations["debug"].attestation,
			attestatations["debug"].certProvider,
			internal.WithAttestationTime(),
			false,
		)

		// then
		assert.ErrorContains(t, err, "attestation was generated in debug mode")
	})
```

Reply:
Done - I also added a test to `test-main`